core/src/main/java/io/substrait/relation/RelChildVisitor.java

```diff
@@ -0,0 +1,275 @@
+package io.substrait.relation;
+
+import io.substrait.expression.Expression;
+import io.substrait.expression.ExpressionVisitor;
+import io.substrait.expression.FieldReference;
+
+public class RelChildVisitor<OUTPUT, EXCEPTION extends Exception>
```

Comment:
Can you add javadocs? Also, it seems the methods return nulls which suggests that we shouldn't have the OUTPUT parameter in the RelVisitor declaration. Correct?

Reply:
Added class level javadoc. Will add more after my next commit as I plan to use the visitor to copy-on-write for simple substitution replacement scenarios.